integr8ly/integreatly-operator
# INTLY-6015 - IDP script improvements for using RHSSO as IDP provider

# Description

IDP script changes to allow setting number of regular users and admin users generated. In addition, dedicated admins now have additional `view-realm` and `manage-users` users in the realm created by the script.

Jira:
* https://issues.redhat.com/browse/INTLY-6015

Related PR:
* https://github.com/RHCloudServices/integreatly-help/pull/434

## Verification
* Install RHMI from this branch
* Follow SOP from PR for installing the IDP with the following changes as it's written for production clusters - https://github.com/RHCloudServices/integreatly-help/pull/434
  * Retrieve offline access token via staging environment 
    * https://qaprodauth.cloud.redhat.com/openshift/token
  * Login to ocm using staging environment url 
    * `ocm login --url=https://api.stage.openshift.com/ --token=<access_token>`
  * Run the IDP script from this PR to generate 1 admin and 1 regular user instead
    * `NUM_ADMIN=1 NUM_REGULAR_USER=1 ./scripts/setup-sso-idp.sh`
* Login to RHSSO using `customer-admin01` user
  * Login using `<rhsso_edge_route>/auth/admin/testing-idp/console`
  * Verify they have access to view only the `testing-idp-realm` and can add users
* Login to RHSSO using `test-user01` user
  * Login using `<rhsso_edge_route>/auth/admin/testing-idp/console`
  * Verify they have access to view and manage their own account

## Type of change




## Checklist
